src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -236,35 +236,33 @@ int fmi3_xml_get_attr_str(fmi3_xml_parser_context_t* context, fmi3_xml_elm_t elm
         return -1;
     }
 
-    *valp = value; /* can be NULL */
+    *valp = value; /* NULL if missing (and not required) */
     return 0;
 }
 
 /**
  * Reads the attribute from attribute buffer as jm_vector(char). This will clear the attribute from the buffer.
+ * @param exists (return arg): If attribute exists (to tell empty string vs missing attribute)
  * @param field (return arg): Attribute value (memory owned by this vector)
  */
 int fmi3_xml_parse_attr_as_string(fmi3_xml_parser_context_t* context, fmi3_xml_elm_t elmID, fmi3_xml_attr_t attrID,
-        int required, jm_vector(char)* field)
+        int required, int* exists, jm_vector(char)* field)
 {
     jm_string val;
     size_t len;
 
     /* Get pointer to attribute value (owned by expat) */
     if (fmi3_xml_get_attr_str(context, elmID, attrID, required, &val)) {
-        return -1;
+        return -1; /* (required and missing) or other error */
     };
 
-    if ((!val || !val[0]) && !required) {
-        /* Return empty string */
-        jm_vector_resize(char)(field, 1);       /* Allocate space for null character */
-        jm_vector_set_item(char)(field, 0, 0);  /* Push null character */
-        jm_vector_resize(char)(field, 0);       /* Make length same as for strlen */
-        return 0;
+    if (!val) { /* not required and missing */
```

Comment:
Difference here:

if val = NULL,  caller gets to decide what to do.

If val != NULL and val[0] == "\0", i.e., empty string, the following constructs are suited to handling this. No extra handling required.